lib/json-string-packer.js

```diff
@@ -0,0 +1,95 @@
+// Copyright IBM Corp. 2014,2016. All Rights Reserved.
+// Node module: loopback-connector
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+var createPromiseCallback = require('./utils').createPromiseCallback;
+
+module.exports = JSONStringPacker;
+
+/**
+ * Create a new Packer instance that can be used to convert between JavaScript
+ * objects and a JsonString representation in a String.
+ *
+ * @param {String} encoding Buffer encoding refer to https://nodejs.org/api/buffer.html#buffer_buffers_and_character_encodings
+ */
+function JSONStringPacker(encoding) {
+  this.encoding = encoding || 'base64';
+}
+
+/**
+ * Encode the provided value to a `JsonString`.
+ *
+ * @param {*} value Any value (string, number, object)
+ * @callback {Function} cb The callback to receive the parsed result.
+ * @param {Error} err
+ * @param {Buffer} data The encoded value
+ * @promise
+ */
+JSONStringPacker.prototype.encode = function(value, cb) {
+  var encoding = this.encoding;
+
+  cb = cb || createPromiseCallback();
+  try {
+    var _toJSON = Buffer.prototype.toJSON;
+
+    Buffer.prototype.toJSON = function() {
```

Comment:
Overwriting a method of a built-in type is a no-go. AFAICT, `JSON.stringify` supports a `replacer` option (see [MDN](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify)), can we use it similarly to how `decode` works?